Lib/test/test_asyncio/test_server.py

```diff
@@ -137,7 +141,8 @@ async def serve(*args):
         self.assertFalse(task1.done())
 
         # active count != 0, not closed: should block
-        srv._attach()
+        addr = srv.sockets[0].getsockname()
+        (rd, wr) = await asyncio.open_connection(addr[0], addr[1])
```

Comment:
Thanks, the calls to `_attach()`/`_detach()` in this test never sat right with me.